include/libint2/cgshell_ordering.h

```diff
@@ -60,13 +60,13 @@ namespace libint2 {
  * am = total angular momentum
  * formula: (am*(am+1))/2 + am+1;
  */
-#define INT_NCART(am) ((((am)+2)*((am)+1))>>1)
+inline int INT_NCART(int am) { return ((am + 2) * (am + 1)) >> 1; }
```

Comment:
thank you for the PR!

These macros are likely used by downstream users so we need to deprecate them first, then remove. This includes some ideas for how to deprecate a macro properly: https://stackoverflow.com/questions/57478368/what-is-the-best-way-to-mark-macro-as-deprecated . Another possibility is to actually replace the macros with functions in global scope and immediately deprecate these functions (although functions cannot be #undef'ed ... hopefully this is not common).

The functions replacing the macros should be namespaced, probably in libint2.

Reply:
> Another possibility is to actually replace the macros with functions in global scope and immediately deprecate these functions

I enclose functions with `namespace libint2` and deprecate the global-scoped correspondences.

> although functions cannot be #undef'ed ... hopefully this is not common

The undefed functions are used locally, so I think the changes have no effect.